node/cmd/ccq/query_server.go

```diff
@@ -0,0 +1,178 @@
+	p2pBootstrap = QueryServerCmd.Flags().String("bootstrap", "", "P2P bootstrap peers (comma-separated)")
+	nodeKeyPath = QueryServerCmd.Flags().String("nodeKey", "", "Path to node key (will be generated if it doesn't exist)")
+	signerKeyPath = QueryServerCmd.Flags().String("signerKey", "", "Path to key used to sign unsigned queries")
+	listenAddr = QueryServerCmd.Flags().String("listenAddr", "[::]:6069", "Listen address for query server (disabled if blank)")
+	permFile = QueryServerCmd.Flags().String("permFile", "", "JSON file containing permissions configuration")
+	ethRPC = QueryServerCmd.Flags().String("ethRPC", "", "Ethereum RPC for fetching current guardian set")
+	ethContract = QueryServerCmd.Flags().String("ethContract", "", "Ethereum core bridge address for fetching current guardian set")
+	logLevel = QueryServerCmd.Flags().String("logLevel", "info", "Logging level (debug, info, warn, error, dpanic, panic, fatal)")
+	telemetryLokiURL = QueryServerCmd.Flags().String("telemetryLokiURL", "", "Loki cloud logging URL")
+	telemetryNodeName = QueryServerCmd.Flags().String("telemetryNodeName", "", "Node name used in telemetry")
+}
+
+var QueryServerCmd = &cobra.Command{
+	Use:   "query-server",
+	Short: "Run the cross-chain query server",
+	Run:   runQueryServer,
+}
+
+func runQueryServer(cmd *cobra.Command, args []string) {
+	common.SetRestrictiveUmask()
+
+	// Setup logging
+	lvl, err := ipfslog.LevelFromString(*logLevel)
+	if err != nil {
+		fmt.Println("Invalid log level")
+		os.Exit(1)
+	}
+
+	logger := ipfslog.Logger("query-server").Desugar()
+	ipfslog.SetAllLoggers(lvl)
+
+	if *telemetryLokiURL != "" {
+		logger.Info("Using Loki telemetry logger")
+		if *telemetryNodeName == "" {
+			logger.Fatal("if --telemetryLokiURL is specified --telemetryNodeName must be specified")
+		}
+		labels := map[string]string{
+			"network":   *p2pNetworkID,
+			"node_name": *telemetryNodeName,
+			"version":   version.Version(),
+		}
+
+		tm, err := telemetry.NewLokiCloudLogger(context.Background(), logger, *telemetryLokiURL, "ccq_server", true, labels)
+		if err != nil {
+			logger.Fatal("Failed to initialize telemetry", zap.Error(err))
+		}
+
+		defer tm.Close()
+		logger = tm.WrapLogger(logger) // Wrap logger with telemetry logger
+	}
+
+	env, err := common.ParseEnvironment(*envStr)
+	if err != nil || (env != common.UnsafeDevNet && env != common.TestNet && env != common.MainNet) {
+		if *envStr == "" {
+			logger.Fatal("Please specify --env")
+		}
+		logger.Fatal("Invalid value for --env, must be dev, test or prod", zap.String("val", *envStr))
+	}
+
+	// Verify flags
+	if *nodeKeyPath == "" {
+		logger.Fatal("Please specify --nodeKey")
+	}
+	if *p2pBootstrap == "" {
+		logger.Fatal("Please specify --bootstrap")
+	}
+	if *permFile == "" {
+		logger.Fatal("Please specify --permFile")
+	}
+	if *ethRPC == "" {
+		logger.Fatal("Please specify --ethRPC")
+	}
+	if *ethContract == "" {
+		logger.Fatal("Please specify --ethContract")
+	}
```

Comment:
Does it make sense to have a check along the lines of:

```go
if *envStr != "dev" && *p2pNetworkID == "/wormhole/dev" {
	logger.Fatal("Please specify either --env=dev or a different --p2pNetworkID")
}
```

At the very minimum there is likely never a case where we want `envStr="prod"` and `p2pNetworkID="/wormhole/dev"` I'd assume. This is a footgun for operators and should be more forgiving. Perhaps we want to hardcode the options for p2pNetworkID?

Reply:
Added a check that the network may not be dev unless the env is also dev. Not putting the check the other way because I'm not sure that wouldn't happen if we are running some sort of ad hoc tool.

Reply:
Perfect sir, thanks!